Add Array filter to search value in `@ORM\Column(type="array")` (#1042)
sonata-project/SonataDoctrineORMAdminBundle: docs/reference/filter_field_definition.rst

For now, only `Doctrine ORM` filters are available:

> * ``Sonata\DoctrineORMAdminBundle\Filter\ArrayFilter``: depends on the ``Sonata\AdminBundle\Form\Type\Filter\ChoiceType`` Form Type,

Reviewer: This is right? Your commit introduces a StringListFilter

Author: Indeed, it's fixed.

        }
    }

ArrayFilter
-----------

> This filter is made to filtering on values saved in databases as a serialized arrays of string with the

Reviewer 2: ```suggestion
This filter is made for filtering on values saved in databases as serialized arrays of strings with the
```

``@ORM\Column(type="array")`` annotation. It is recommended to use another table and ``OneToMany`` relations

> if you want to make complex ``SQL`` queries or if your table is too big and you get performance issue but

Reviewer 2: ```suggestion
if you want to make complex ``SQL`` queries or if your table is too big and you get performance issues but
```

this filter can provide some basic queries::

    protected function configureDatagridFilters(DatagridMapper $datagridMapper): void
    {
        $datagridMapper
            ->add('keywords', ArrayFilter::class);

>     }

Reviewer: Please give some examples how to fill the filter. What are correct user inputs

Author: I updated the example, is it ok for you now ?

.. note::

    The filter can give bad results with associative arrays since it is not easy to distinguish between keys

>     and values for a serialized associative array.

Reviewer 2: Add also a warning about values that are `s` or `b` or whatever can confuse this? BTW, maybe this would be less risky with json…

Author: I don't understand why `s` or `b` can confuse this @greg0ire. 
I serialize the value you're looking for. So if you're looking for `s`, it'll search
`"s:1:"s";"` this can't be confused with a `s` in a bigger word or the `s` used by the serialize array to say `string`.

The only possible confusion is when you store string which are already contain serialize value like storing the array `['s:4:"toto";', 'toto']`. But this seems to be a really weird case. Should I mention it ?

It does not seems to me easy to migrate a `@ORM\Column(type="array")` to a `@ORM\Column(type="json")`, isn't it ? So I would prefer to offer a filter for `array` and a filter for `json` (or the same filter with a `type` option).

Reviewer 2: Oh right, `s` and `b` look safe to search for indeed, my bad.
